stores/cachestore/cache_store.go

```diff
@@ -0,0 +1,49 @@
+package cachestore
+
+import (
+	"os"
+	"time"
+
+	"github.com/mailchain/mailchain/encoding"
+	"github.com/mailchain/mailchain/internal/hash"
+
+	"github.com/spf13/afero"
+)
+
+type CacheStore struct {
+	cache afero.Fs
+}
+
+func NewCacheStore(cacheTimeout time.Duration) *CacheStore {
+	base := afero.NewOsFs()
```

Comment:
looks like the files will be store in the working path which is not the desired behaviour. Try something like this `base := afero.NewBasePathFs(afero.NewOsFs(), path)` path will need to be passed in through the config